week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/baseProduct.ts

```diff
@@ -0,0 +1,4 @@
+export class BaseProduct {
```

Comment:
Tutaj możesz się bardziej ubezpieczyć.
```typescript
export abstract class BaseProduct
```

oraz dodać pole:
```typescript
abstract public type: ProductType
````

Polecam Ci poczytać o koncepcji klas absrakcyjnych i pól / metod też.

`abstract` w klasie powoduje że NIE MOŻESZ zrobić tak:

```typescript
const instance = new BaseProduct(....)
````

Więc Ty celowo pokazujesz - że jeśli ma istnieć jakiś produkt - to musi być klasa która dziedziczy po `BaseProduct`

z kolei `abstract` przy polu - pozwala je zdefiniować - bez podania jego wartości (coś jak interface)
Natomiast WYMUSZA jego zdefiniowanie przy dziedziczeniu przez klasy.